src/resolve/resolved-llmnr.c

```diff
@@ -134,58 +134,90 @@ int manager_llmnr_ipv4_udp_fd(Manager *m) {
         if (m->llmnr_ipv4_udp_fd >= 0)
                 return m->llmnr_ipv4_udp_fd;
 
-        m->llmnr_ipv4_udp_fd = socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
-        if (m->llmnr_ipv4_udp_fd < 0)
+        r = socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (r < 0) {
+                log_error_errno(r, "LLMNR-IPv4(udp): Failed to create socket: %m");
                 return -errno;
+        }
+        m->llmnr_ipv4_udp_fd = r;
 
         /* RFC 4795, section 2.5 recommends setting the TTL of UDP packets to 255. */
         r = setsockopt(m->llmnr_ipv4_udp_fd, IPPROTO_IP, IP_TTL, &ttl, sizeof(ttl));
         if (r < 0) {
+                log_error_errno(r, "LLMNR-IPv4(udp): Failed to set IP_TTL: %m");
                 r = -errno;
                 goto fail;
         }
 
         r = setsockopt(m->llmnr_ipv4_udp_fd, IPPROTO_IP, IP_MULTICAST_TTL, &ttl, sizeof(ttl));
         if (r < 0) {
+                log_error_errno(r, "LLMNR-IPv4(udp): Failed to set IP_MULTICAST_TTL: %m");
```

Comment:
Why lowercase: "udp" not "UDP"? We try to use correct capitalization of abbreviations where possible.